### ISSUE TITLE
Specify layer name for writing to FileGDB?

### ISSUE THREAD

**Issue by DanCranford** [duckdb/duckdb_spatial#265]
Hey there - 

Is it possible to specify a layer name (feature class or table) when writing to a file geodatabase?  

The following code creates a file geodatabase named "testing" and a feature class named "testing".
```
COPY 
    (SELECT ST_Point(1,2) as geom, 10 as i) 
TO 
    './testing.gdb' 
WITH (FORMAT 'GDAL', 
      DRIVER 'OpenFileGDB', 
      GEOMETRY_TYPE 'POINT', 
      LAYER_CREATION_OPTIONS 'LAYER_ALIAS=alias');
```

It appears, based on the [OpenFileGDB driver documentation](https://gdal.org/drivers/vector/openfilegdb.html) from GDAL, that you can specify the file geodatabase and feature class name when copying.

Thanks!

**Comment by Maxxen** [comment]
What you are looking for is passing the `layer_name` argument like this
```sql
COPY 
    (SELECT ST_Point(1,2) as geom, 10 as i) 
TO 
    './testing.gdb' 
WITH (FORMAT 'GDAL', 
      DRIVER 'OpenFileGDB', 
      GEOMETRY_TYPE 'POINT',
      LAYER_NAME 'My layer name');
```

The `LAYER_ALIAS` only sets the alias in the metadata. Im not actually sure what that does since I don't use any ESRI products, but its not something we make use of when reading filegdbs anyway.

**Comment by DanCranford** [comment]
Thanks @Maxxen!  That's exactly what I was looking for.  Is it possible to write to an existing file geodatabase?  I've noticed that if I try to create two datasets in the same database, I get the following error.


```
IOException                               Traceback (most recent call last)
~\AppData\Local\Temp\1\ipykernel_14812\3340377106.py in <cell line: 1>()
----> 1 conn.execute("""COPY 
      2     (SELECT ST_Point(1,2) as geom, 10 as i)
      3 TO
      4     './gdb_name.gdb'
      5 WITH (FORMAT 'GDAL', 

IOException: IO Error: GDAL Error (1): ./gdb_name.gdb already exists.
```

**Comment by Maxxen** [comment]
No, I don't think this is supported yet, but it might be possible to implement in the future.

**Comment by rhipnox** [comment]
Hi and thanks again for all the great work on DuckDB and the spatial extension — it's been a game changer for our workflows. Being able to handle spatial data with SQL, lightweight setup, and fast performance is simply awesome 🚀

I just wanted to check in and see if there's been any progress regarding support for adding multiple tables to an existing `.gdb` file. Right now, we can write the first layer just fine, but any subsequent writes throw a `IOException: IO Error: GDAL Error (1): ./gdb_name.gdb already exists.` error, as mentioned above.

We rely on `.gdb` export to integrate with ArcGIS workflows, and supporting multiple layers would make things much smoother.

Totally understand if it's not a high priority — just wanted to express how useful it would be. And of course, happy to help test if needed!

Thanks again for pushing spatial capabilities forward in the DuckDB ecosystem 🙌

**Comment by rhipnox** [comment]
Hey @Maxxen !

Apologies for the disturbance. Is there any update or roadmap for this _already exists_ issue?

**Comment by pinkerltm** [comment]
Hi Folks! We also would very much appreciate if we can make this working. Surely we try to promote more commonly usable formats, but if we as solution provider want to be taken serious, we still have to integrate the ESRI users as well. So +1 for this one.

**Comment by Maxxen** [comment]
I think for formats like this it would be better for us to implement a "storage extension" so that you could `ATTACH` to a filegdb and insert/read from the layers as normal tables, instead of relying on ST_Read.

We're currently redoing a lot of stuff in core DuckDB to make extensions more powerful and easier to maintain, and once we have a better abstraction for extension storages I would be very excited to add this to spatial.
